app/controllers/actions/IdentifierAction.scala

```diff
@@ -39,14 +41,28 @@ class AuthenticatedIdentifierAction @Inject() (
     extends IdentifierAction
     with AuthorisedFunctions {
 
+  val enrolmentKey: String = config.enrolmentKey
+
   override def invokeBlock[A](request: Request[A], block: IdentifierRequest[A] => Future[Result]): Future[Result] = {
 
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
 
-    authorised().retrieve(Retrievals.internalId) {
-      _.map {
-        internalId => block(IdentifierRequest(request, internalId))
-      }.getOrElse(throw new UnauthorizedException("Unable to retrieve internal Id"))
+    authorised().retrieve(Retrievals.internalId and Retrievals.allEnrolments and affinityGroup and credentialRole) {
+      case _ ~ enrolments ~ _ ~ _ =>
+        if (enrolments.enrolments.exists(_.key == enrolmentKey)) {
+          Future.successful(Redirect(config.crsFatcaFIManagementFrontendUrl))
+        } else {
+          Future.successful(Redirect(routes.UnauthorisedStandardUserController.onPageLoad()))
```

Comment:
@AashuChauhanHmrc This is not the correct behaviour. This will show all users the unauthorised page.